### ISSUE TITLE
End offset for text containing entities is incorrect

### ISSUE THREAD

**Issue by siefkenj** [syntax-tree/xast-util-from-xml#3]
### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.1

### Steps to reproduce

Exectue `fromXml("<foo>text&amp;text</foo>")`

### Expected behavior

The `offset` position should start at `5` and end at `18`

### Actual behavior

The `offset` position should start at `5` and end at `15`

### Affected runtime and version

node@18.12.1

**Comment by wooorm** [comment]
Hmm, the positional info of `sax` isn’t very good.
I’m not a giant fan of `sax`, it’s very loose, pretty big.

I think I’d like to switch to something else. Perhaps https://github.com/rgrove/parse-xml. Seems fast. Halve the size. Well-maintained and modern. Doesn’t expose positional info from what I can see though

**Comment by siefkenj** [comment]
Another parser is probably a good idea. I encountered this error when working on a converter from [lezer-xml](https://github.com/lezer-parser/xml) to `xast`.

I have it basically working: https://github.com/siefkenj/typescript-xml/tree/main/packages/xml/src/lezer/lezer-to-xast

The only differences between `fromXml` and what I just made are correct positions and no errors (lezer is designed to parse no matter what, though it can be instructed to throw on the first error).
